pydropsonde/circles.py

```diff
@@ -147,16 +153,57 @@ def get_xy_coords_for_circles(self):
 
         return self
 
+    def apply_interp_na(self, method="cubic", max_gap=1500, x_method="cubic"):
+        variables = ["u", "v", "q", "ta", "p"]
+        dss = {}
+        ds = self.circle_ds
+        alt_dim = self.alt_dim
+        ds["p"] = np.log(ds["p"])
+        if method is not None:
+            ds_x = ds.x.interpolate_na(
+                dim=alt_dim,
+                method=x_method,
+                bounds_error=False,
+                fill_value=np.nan,
+                max_gap=int(max_gap),
+            )
+            ds_y = ds.y.interpolate_na(
+                dim=alt_dim,
+                method=x_method,
+                bounds_error=False,
+                fill_value=np.nan,
+                max_gap=int(max_gap),
+            )
```

Comment:
Not sure whether it's elegant, but I was wondering whether you could reduce the code duplication by writing the interpolated x and y directly to the new `dss`?

```suggestion
for xy in ["x", "y"]:
    dss[xy] = ds[xy].interpolate_na(
                dim=alt_dim,
                method=x_method,
                bounds_error=False,
                fill_value=np.nan,
                max_gap=int(max_gap),
            )
```

Reply:
as above: There is only one call now  :)